src/org/omegat/util/gui/FontFallbackManager.java

```diff
@@ -52,15 +57,117 @@ private FontFallbackManager() {
      * <li><a href="https://github.com/googlei18n/noto-emoji">Noto Emoji</a>
      * </ul>
      */
-    private static final Set<String> FONT_BLACKLIST = Collections.singleton("Apple Color Emoji");
-    private static final Font FONT_UNAVAILABLE = new Font("", 0, 0);
+    private static final Set<String> FONT_BLACKLIST = new HashSet<>(
+            Arrays.asList("Apple Color Emoji", "Noto Color Emoji")
+    );
+    private static final Font FONT_UNAVAILABLE = new Font("", Font.PLAIN, 0);
 
     private static final Logger LOGGER = Logger.getLogger(FontFallbackManager.class.getName());
 
     private static final Font[] RECENT_FONTS = new Font[8];
     private static int lastFontIndex = 0;
     private static final Map<Integer, Font> CACHE = new ConcurrentHashMap<>();
 
+    private static final String BOLD_SUFFIX = ".bold";
+    private static final String ITALIC_SUFFIX = ".italic";
+    private static final List<String> myFontFamilyNames = new ArrayList<>();
+
+    private static final FontRenderContext DEFAULT_CONTEXT = new FontRenderContext(null, false, false);
+
+    static {
+        String[] fontFamilies = GraphicsEnvironment.getLocalGraphicsEnvironment().getAvailableFontFamilyNames();
+        for (final String fontName : fontFamilies) {
+            if (!fontName.endsWith(BOLD_SUFFIX) && !fontName.endsWith(ITALIC_SUFFIX)) {
+                myFontFamilyNames.add(fontName);
+            }
+        }
+    }
```

Comment:
A couple things:

1. I don't understand why this moved to `FontFallbackManager`. Callers of `StaticUtils#getFontNames` are not interested in font fallback, and neither `FontFallbackManager#getFontNames` nor `myFontFamilyNames` are used within `FontFallbackManager` at all. I think it should be moved back to `StaticUtils`.
2. None of the callers of `getFontNames` are in a hot loop, so I am skeptical that it is meaningful performance-wise to cache the family names. In fact by caching the names but never invalidating the cache, you prevent new fonts from being picked up (I don't actually know if this API actually can pick up newly installed fonts, but if it can then your caching scheme definitely breaks that).
3. What is the `BOLD_SUFFIX` and `ITALIC_SUFFIX`? For reference on my machine (macOS 11.4) none of my fonts have such suffixes:

    <details><summary>Aaron's fonts</summary>
    [.AppleSystemUIFont, Academy Engraved LET, Agave, Al Bayan, Al Nile, Al Tarikh, Alegreya Sans, Alegreya Sans SC, all-the-icons, American Typewriter, Andale Mono, Anonymous Pro, Apple Braille, Apple Chancery, Apple Color Emoji, Apple LiGothic, Apple LiSung, Apple SD Gothic Neo, Apple Symbols, AppleGothic, AppleMyungjo, Arial, Arial Black, Arial Hebrew, Arial Hebrew Scholar, Arial Narrow, Arial Rounded MT Bold, Arial Unicode MS, Avenir, Avenir Next, Avenir Next Condensed, Ayuthaya, BabelStone Han, BabelStone Han PUA, Baghdad, Baloo, Baloo Bhai, Baloo Bhaijaan, Baloo Bhaina, Baloo Chettan, Baloo Da, Baloo Paaji, Baloo Tamma, Baloo Tammudu, Baloo Thambi, Bangla MN, Bangla Sangam MN, Baoli SC, Baoli TC, Baskerville, Beirut, BiauKai, Big Caslon, Bodoni 72, Bodoni 72 Oldstyle, Bodoni 72 Smallcaps, Bodoni Ornaments, Bradley Hand, Brush Script MT, Cambay Devanagari, Cascadia Code, Cascadia Code Italic, Cascadia Code PL, Cascadia Code PL Italic, Cascadia Mono, Cascadia Mono Italic, Cascadia Mono PL, Cascadia Mono PL Italic, Chalkboard, Chalkboard SE, Chalkduster, Charter, Cochin, Comic Sans MS, Copperplate, Corsiva Hebrew, Courier, Courier New, Damascus, DecoType Naskh, Devanagari MT, Devanagari Sangam MN, Dialog, DialogInput, Didot, DIN Alternate, DIN Condensed, Diwan Kufi, Diwan Thuluth, EB Garamond, EB Garamond 12 All SC, EB Garamond Initials, EB Garamond Initials Fill1, EB Garamond Initials Fill2, EB Garamond SC, Envy Code R, ETBembo, ETBookOT, Euphemia UCAS, Farah, Farisi, file-icons, Fira Code, FiraGO, FontAwesome, Futura, Galvji, GB18030 Bitmap, Geeza Pro, Geneva, Georgia, Gill Sans, github-octicons, Go, Go Medium, Go Mono, Go Smallcaps, Gotu, Grantha Sangam MN, Gujarati MT, Gujarati Sangam MN, GungSeo, Gurmukhi MN, Gurmukhi MT, Gurmukhi Sangam MN, Hack, HanaMinA, HanaMinB, Hangulatin EN, Hannotate SC, Hannotate TC, HanziPen SC, HanziPen TC, HeadLineA, Hei, Heiti SC, Heiti TC, Helvetica, Helvetica Neue, Herculanum, Hiragino Maru Gothic ProN, Hiragino Mincho ProN, Hiragino Sans, Hiragino Sans CNS, Hiragino Sans GB, Hoefler Text, IBM 3270, IBM 3270 Condensed, IBM 3270 Semi-Condensed, IBM Plex Mono, IBM Plex Sans, IBM Plex Sans Arabic, IBM Plex Sans Condensed, IBM Plex Sans Devanagari, IBM Plex Sans Hebrew, IBM Plex Sans KR, IBM Plex Sans Thai, IBM Plex Sans Thai Looped, IBM Plex Serif, Impact, InaiMathi, Inconsolata, Input Mono, Input Mono Compressed, Input Mono Condensed, Input Mono Narrow, Input Sans, Input Sans Compressed, Input Sans Condensed, Input Sans Narrow, Input Serif, Input Serif Compressed, Input Serif Condensed, Input Serif Narrow, Inter, Iosevka, ITF Devanagari, ITF Devanagari Marathi, Jaini, Jaini Purva, JetBrains Mono, JetBrains Mono NL, JuliaMono, Kai, Kailasa, Kaiti SC, Kaiti TC, Kannada MN, Kannada Sangam MN, Kefa, Khmer MN, Khmer Sangam MN, Klee, Kohinoor Bangla, Kohinoor Devanagari, Kohinoor Gujarati, Kohinoor Telugu, Kokonor, Krungthep, KufiStandardGK, Lahore Gurmukhi, Lantinghei SC, Lantinghei TC, Lao MN, Lao Sangam MN, Libian SC, Libian TC, Ligconsolata, LiHei Pro, LingWai SC, LingWai TC, LiSong Pro, Lucida Grande, Luminari, M+ 1c, M+ 1m, M+ 1mn, M+ 1p, M+ 2c, M+ 2m, M+ 2p, Maku, Malayalam MN, Malayalam Sangam MN, Marker Felt, Material Icons, MD IO 0.2, Menlo, Meta, Microsoft Sans Serif, MigMix 1M, MigMix 1P, MigMix 2M, MigMix 2P, Migu 1C, Migu 1M, Migu 1P, Migu 2M, Mishafi, Mishafi Gold, Modak, Monaco, mononoki, Monospaced, Mshtakan, Mukta, Mukta Mahee, Mukta Malar, Mukta Vaani, Muna, Myanmar MN, Myanmar Sangam MN, Nadeem, Nanum Brush Script, Nanum Gothic, Nanum Myeongjo, Nanum Pen Script, Neue Haas Grotesk Text Pro, New Peninim MT, Noteworthy, Noto Nastaliq Urdu, Noto Sans Kannada, Noto Sans Myanmar, Noto Sans Oriya, Noto Serif Kannada, Noto Serif Myanmar, Optima, Oriya MN, Oriya Sangam MN, Osaka, Palatino, Papyrus, Party LET, PCMyungjo, Phosphate, Piazzolla, Piazzolla SC, PilGi, PingFang HK, PingFang SC, PingFang TC, Plantagenet Cherokee, PragmataPro, PragmataPro Liga, PragmataPro Mono, PragmataPro Mono Liga, PSL Ornanong Pro, PT Mono, PT Sans, PT Sans Caption, PT Sans Narrow, PT Serif, PT Serif Caption, Raanana, Rockwell, Sana, SansSerif, Sarasa Fixed CL, Sarasa Fixed HC, Sarasa Fixed J, Sarasa Fixed K, Sarasa Fixed SC, Sarasa Fixed Slab CL, Sarasa Fixed Slab HC, Sarasa Fixed Slab J, Sarasa Fixed Slab K, Sarasa Fixed Slab SC, Sarasa Fixed Slab TC, Sarasa Fixed TC, Sarasa Gothic CL, Sarasa Gothic HC, Sarasa Gothic J, Sarasa Gothic K, Sarasa Gothic SC, Sarasa Gothic TC, Sarasa Mono CL, Sarasa Mono HC, Sarasa Mono J, Sarasa Mono K, Sarasa Mono SC, Sarasa Mono Slab CL, Sarasa Mono Slab HC, Sarasa Mono Slab J, Sarasa Mono Slab K, Sarasa Mono Slab SC, Sarasa Mono Slab TC, Sarasa Mono TC, Sarasa Term CL, Sarasa Term HC, Sarasa Term J, Sarasa Term K, Sarasa Term SC, Sarasa Term Slab CL, Sarasa Term Slab HC, Sarasa Term Slab J, Sarasa Term Slab K, Sarasa Term Slab SC, Sarasa Term Slab TC, Sarasa Term TC, Sarasa UI CL, Sarasa UI HC, Sarasa UI J, Sarasa UI K, Sarasa UI SC, Sarasa UI TC, Sathu, Savoye LET, Sawarabi Mincho, Serif, SF Compact, SF Compact Display, SF Compact Rounded, SF Compact Text, SF Pro, SF Pro Display, SF Pro Rounded, SF Pro Text, Shobhika, Shree Devanagari 714, Signifier Test, SignPainter, Silom, SimSong, Sinhala MN, Sinhala Sangam MN, Skia, Snell Roundhand, Songti SC, Songti TC, Source Sans 3, STFangsong, STHeiti, STIXGeneral, STIXIntegralsD, STIXIntegralsSm, STIXIntegralsUp, STIXIntegralsUpD, STIXIntegralsUpSm, STIXNonUnicode, STIXSizeFiveSym, STIXSizeFourSym, STIXSizeOneSym, STIXSizeThreeSym, STIXSizeTwoSym, STIXVariants, STKaiti, STSong, Sudo, Sudo UI, Sudo Var, Sukhumvit Set, Symbol, Tahoma, Tamil MN, Tamil Sangam MN, Telugu MN, Telugu Sangam MN, Thonburi, Times, Times New Roman, Tiro Bangla, Tiro Gurmukhi, Tiro Hindi, Tiro Kannada, Tiro Marathi, Tiro Sanskrit, Tiro Tamil, Tiro Telugu, Toppan Bunkyu Gothic, Toppan Bunkyu Midashi Gothic, Toppan Bunkyu Midashi Mincho, Toppan Bunkyu Mincho, Trattatello, Trebuchet MS, Tsukushi A Round Gothic, Tsukushi B Round Gothic, Unifont, Unifont-JP, Valkyrie C3, Valkyrie C3 Tab, Valkyrie C4, Valkyrie C4 Tab, Valkyrie T3, Valkyrie T3 Tab, Valkyrie T4, Valkyrie T4 Tab, Verdana, Victor Mono, Waseem, Wawati SC, Wawati TC, Weather Icons, Webdings, Weibei SC, Weibei TC, Wingdings, Wingdings 2, Wingdings 3, Xingkai SC, Xingkai TC, Yuanti SC, Yuanti TC, YuGothic, YuKyokasho, YuKyokasho Yoko, YuMincho, YuMincho +36p Kana, Yuppy SC, Yuppy TC, Zapf Dingbats, Zapfino]
    </details>

Reply:
> None of the callers of getFontNames are in a hot loop,
This is regression when porting from my local dev branch to here.

> What is the BOLD_SUFFIX and ITALIC_SUFFIX?

These idea come from intellij-community source. I also don't know in my platform. So i'd like to remove it.